keras_retinanet/preprocessing/generator.py

```diff
@@ -43,6 +43,7 @@ def __init__(
         image_min_side=600,
         image_max_side=1024,
         transform_parameters=None,
+        model=None,
```

Comment:
I'm not a big fan of passing the model here. Let's discuss this in the comment about `true-shapes`.

Reply:
Not a fan of coupling generators with models. Those two should be completely independent in my mind.

Reply:
As a matter of principle, I do agree with that, the generator should be separate from the model. On the other hand, the generator does need to somehow be aware of how the model backbone expects its data. Moreover, the shape of the anchors depends on the input shape so (at best) it could be passed as a callback, but the callback itself would then be computed from the model, so it's almost the same thing, but a bit more complicated.

Reply:
At the very least the callback will keep the generator API clean, so I think we should do that.

It would also give the option to determine the shapes from some other formula instead of extracting them from the model. I'm assuming there must be some logic to it.